rules/numeric-separators-style.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+const merge = require('lodash/merge');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+function addSeparator(value, {minimumDigits, groupLength}, fromLeft) {
+	const {length} = value;
+
+	if (length < minimumDigits) {
+		return value;
+	}
+
+	const parts = [];
+	if (fromLeft) {
+		for (let start = 0; start < length; start += groupLength) {
+			const end = Math.min(start + groupLength, length);
+			parts.push(value.slice(start, end));
+		}
+	} else {
+		for (let end = length; end > 0; end -= groupLength) {
+			const start = Math.max(end - groupLength, 0);
+			parts.unshift(value.slice(start, end));
+		}
+	}
+
+	return parts.join('_');
+}
+
+function addSeparatorFromLeft(value, options) {
+	return addSeparator(value, options, true);
+}
+
+function formatNumber(value, options) {
+	const parts = value.split('.');
+	const [decimal, fractional] = parts;
```

Comment:
`decimal` seems should name as `integer`, sorry for my bad english.

Reply:
No problem ^^